subprojects/gradle/buildchecks/src/test/kotlin/com/avito/android/plugin/build_param_check/incremental_check/IncrementalKaptTaskTest.kt

```diff
@@ -0,0 +1,126 @@
+package com.avito.android.plugin.build_param_check.incremental_check
+
+import com.avito.test.gradle.AndroidAppModule
+import com.avito.test.gradle.TestProjectGenerator
+import com.avito.test.gradle.gradlew
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+internal class IncrementalKaptTaskTest {
+
+    private lateinit var projectDir: File
+
+    @BeforeEach
+    internal fun setUp(@TempDir projectDir: File) {
+        this.projectDir = projectDir
+    }
+
+    @Test
+    fun `build success with no warnings - unsupported Java version`() {
```

Comment:
You could try to use the Junit5 test generator.
It will reduce boilerplate for writing the same test bodies.

Reply:
I don't think that 2 lines of code are boilerplate. BTW these tests check different situations (success - fail, warning - no log) and it will become parameters hell if I try to use test generator.